ExtremeValueProject/AffineTransformation.lean

```diff
@@ -322,14 +322,41 @@ lemma AffineEquiv.extend_top' (A : ℝ ≃ᵃ[ℝ] ℝ) :
 --    A.toAffineMap.extend x = A x :=
 --  rfl
 
+lemma AffineEquiv.extend_symm_cancel (A : ℝ ≃ᵃ[ℝ] ℝ) (x : EReal) :
+    A.symm.toAffineMap.extend (A.toAffineMap.extend x) = x ∧
+    A.toAffineMap.extend (A.symm.toAffineMap.extend x) = x := by
```

Comment:
I think instead of stating the conjunction `... ∧ ...`, it is better to state just one of the two. The statement becomes cleaner, the proof maybe a few lines shorter, and one can later apply either `extend_symm_cancel A x` or `extend_symm_cancel A.symm x` to get something that is definitionally equal (I believe) to whichever direction one needs.

The following suggestion will break the proof you provided, but it should be easy to fix by just removing the parts which work on half of the goal.

```suggestion
lemma AffineEquiv.extend_symm_cancel (A : ℝ ≃ᵃ[ℝ] ℝ) (x : EReal) :
    A.toAffineMap.extend (A.symm.toAffineMap.extend x) = x := by
```

Reply:
Thanks! Your proof is ok, but I left one comment on an improvement.

Reply:
Thanks for the tip! I decided not to add an explicit `symm_extend_cancel` since `extend_symm_cancel` can already be used in the other direction so nicely.